src/NeoSharp.Core/Messaging/Handlers/VersionMessageHandler.cs

```diff
@@ -26,12 +41,16 @@ public VersionMessageHandler(IServer server, ILogger<VersionMessageHandler> logg
                 throw new InvalidOperationException($"The handshake is failed due to \"{nameof(_server.Version.Nonce)}\" value equality.");
             }
 
+            // Send Ack
+
+            await sender.Send<VerAckMessage>();
```

Comment:
Is it better to answer on changed protocol?

Reply:
I think that we need to create a new command, for change protocol

Reply:
My thinking is HTTP. When the server sends HTTP headers to the client it states max supported version and it is up to client to say no and send back lower for server to switch. I am more towards not to introduce extra command to handshake faster. Indeed, we are not going to change protocol after handshake is done.

Reply:
Is necessary change the protocol in both sides, if the other side don't know your version or your intention to change, is not possible change it